tree: compare Geometry/Geography types using protobuf bytes

otan opened the pull request:
We previously used EWKB() to compare protobuf bytes, but we are looking
to move away from storing EWKB as the backing store for performance
reasons.

As such, change the comparisons to use protobuf compares.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50624)
<!-- Reviewable:end -->

mjibson commented on pkg/geo/geo.go:
```diff
@@ -638,3 +640,17 @@ func shapeFromGeom(t geom.T) (geopb.Shape, error) {
 }
+
+// CompareSpatialObject compares the SpatialObject.
+// This should be consistent with the way it is stored.
```
Can this line be more specific? Maybe `This must match the byte ordering that is be produced by encoding.EncodeGeoAscending.`

otan committed:
tree: compare Geometry/Geography types using protobuf bytes

We previously used EWKB() to compare protobuf bytes, but we are looking
to move away from storing EWKB as the backing store for performance
reasons.

As such, change the comparisons to use protobuf compares.

Release note: None

otan commented:
bors r=mjibson
